components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -2005,6 +2008,137 @@ void JsonRpcService::GetERC1155TokenBalance(
       network_url, std::move(internal_callback));
 }
 
+void JsonRpcService::DiscoverAssets(
+    const std::string& chain_id,
+    const std::vector<std::string>&
+        account_addresses,  // TODO(nvonpentz) - Use AccountInfoPtr instead
+    std::vector<mojom::BlockchainTokenPtr> user_assets,
+    DiscoverAssetsCallback callback) {
+  if (chain_id != mojom::kMainnetChainId) {
```

Comment:
What's the motivation of limiting asset discovery only to mainnet for now?

Reply:
I wasn't sure if we had complete token registries for other chains.

Also this depends on fast eth_getLogs queries - https://blog.infura.io/post/faster-logs-and-events-e43e2fa13773.  I'm not sure if that's implemented for other chains.